--- a/1/main.fractch
+++ b/1/main.fractch
@@ -36,12 +36,12 @@
 costume "part 7//1" file "assets/part_7__1.svg" center 86.3143032878551,106.6261518675913;
 costume "part 7//2" file "assets/part_7__2.svg" center 86.3143032878551,106.62609186759133 current;
 
-when broadcast "intro frame" at 180,331 {
-  setSize 166;
-  gotoXY -260, -14;
-  broadcast "part 1";
+on message "intro frame" at 180,331 {
+  self.size = 166;
+  self.position = (-260, -14);
+  emit "part 1";
   looks.switchcostumeto(COSTUME: shadow looks.costume(field COSTUME: ["part 1//1","2u8}sry5VgDFoC1]:.%*"]));
-  glideXY 0.6, 0, -14;
+  self.glideTo(0, -14, 0.6);
   looks.switchcostumeto(COSTUME: shadow looks.costume(field COSTUME: ["part 1//2","fVhl(PC@ELVkkkrK}WiK"]));
   wait 0.1;
   looks.switchcostumeto(COSTUME: shadow looks.costume(field COSTUME: ["part 1//3","tHY1Z!{L!LjRl(*mG6nh"]));
@@ -58,8 +58,8 @@
   wait 0.2;
 }
 
-when broadcast frame1 at 517,299 {
-  broadcast "part 2";
+on message frame1 at 517,299 {
+  emit "part 2";
   looks.switchcostumeto(COSTUME: shadow looks.costume(field COSTUME: ["part 2//1","x*I7Qx:^AN9c#n{6Dvi5"]));
   wait 0.2;
   looks.switchcostumeto(COSTUME: shadow looks.costume(field COSTUME: ["part 2//2","WBJJ6D,}nWC@E#~U*=3|"]));
@@ -70,8 +70,8 @@
   wait 0.1;
 }
 
-when broadcast frame2 at 888,346 {
-  broadcast "part 3";
+on message frame2 at 888,346 {
+  emit "part 3";
   looks.switchcostumeto(COSTUME: shadow looks.costume(field COSTUME: ["part 3//1","OPcUYI8)7bW,aiFItihD"]));
   wait 0.2;
   looks.switchcostumeto(COSTUME: shadow looks.costume(field COSTUME: ["part 3//2","1g86!.N)WImq#qtJjSLT"]));
@@ -82,8 +82,8 @@
   wait 0.1;
 }
 
-when broadcast frame3 at 1259,314 {
-  broadcast "part 4";
+on message frame3 at 1259,314 {
+  emit "part 4";
   looks.switchcostumeto(COSTUME: shadow looks.costume(field COSTUME: ["part 4//1","p,H~gQ/z-j:rT44c_/1!"]));
   wait 0.3;
   looks.switchcostumeto(COSTUME: shadow looks.costume(field COSTUME: ["part 4//2","jgqyp:]FZO+Im@H8sdG2"]));
@@ -96,8 +96,8 @@
   wait 0.3;
 }
 
-when broadcast frame4 at 531,888 {
-  broadcast "part 5";
+on message frame4 at 531,888 {
+  emit "part 5";
   looks.switchcostumeto(COSTUME: shadow looks.costume(field COSTUME: ["part 5//Taco","MB-0rW;WZy2/4=UdeNII"]));
   wait 0.2;
   looks.switchcostumeto(COSTUME: shadow looks.costume(field COSTUME: ["part 5//Taco3","K72RiB~f0f5Xr)Nxu8Gm"]));
@@ -108,8 +108,8 @@
   wait 0.2;
 }
 
-when broadcast frame5 at 942,901 {
-  broadcast "part 6";
+on message frame5 at 942,901 {
+  emit "part 6";
   looks.switchcostumeto(COSTUME: shadow looks.costume(field COSTUME: ["part 6 Wrong//1","yLvyKH9,LpfM(!1?o+y@"]));
   wait 0.2;
   looks.switchcostumeto(COSTUME: shadow looks.costume(field COSTUME: ["part 6 Wrong//2","-|=A`zzHp%d9V3-BbByo"]));
@@ -124,8 +124,8 @@
   wait 0.2;
 }
 
-when broadcast frame6 at 1347,1029 {
-  broadcast "part 7";
+on message frame6 at 1347,1029 {
+  emit "part 7";
   looks.switchcostumeto(COSTUME: shadow looks.costume(field COSTUME: ["part 7//1","I0-q?-y=N(K5/W}+VTDG"]));
   wait 0.2;
   looks.switchcostumeto(COSTUME: shadow looks.costume(field COSTUME: ["part 7//2",")6(3=;o-cSg`?[OC^`io"]));
--- a/13/main.fractch
+++ b/13/main.fractch
@@ -14,30 +14,30 @@
 sound "part 6" file "assets/part_6.wav" rate 48000 samples 90164;
 sound "part 7" file "assets/part_7.wav" rate 48000 samples 37388;
 
-when broadcast "part 1" at 253,351 {
+on message "part 1" at 253,351 {
   sound.playuntildone(SOUND_MENU: shadow sound.sounds_menu(field SOUND_MENU: ["part 1","Fe:6MfRw4dE{:qO?cx%M"]));
 }
 
-when broadcast "part 2" at 594,351 {
+on message "part 2" at 594,351 {
   sound.playuntildone(SOUND_MENU: shadow sound.sounds_menu(field SOUND_MENU: ["part 2","t8m5yl./c=S;8I:3z*Id"]));
 }
 
-when broadcast "part 3" at 912,354 {
+on message "part 3" at 912,354 {
   sound.playuntildone(SOUND_MENU: shadow sound.sounds_menu(field SOUND_MENU: ["part 3","UH~;iU/%kcqVNq86NZZm"]));
 }
 
-when broadcast "part 4" at 1266,359 {
+on message "part 4" at 1266,359 {
   sound.playuntildone(SOUND_MENU: shadow sound.sounds_menu(field SOUND_MENU: ["part 4","z(J%3/7GHQ]Z)iL9xw[("]));
 }
 
-when broadcast "part 5" at 210,641 {
+on message "part 5" at 210,641 {
   sound.playuntildone(SOUND_MENU: shadow sound.sounds_menu(field SOUND_MENU: ["part 5","{up(M;/NLOOW8!/+NV2f"]));
 }
 
-when broadcast "part 6" at 566,634 {
+on message "part 6" at 566,634 {
   sound.playuntildone(SOUND_MENU: shadow sound.sounds_menu(field SOUND_MENU: ["part 6","+uphy6x;D79E;Y6XaomS"]));
 }
 
-when broadcast "part 7" at 940,638 {
+on message "part 7" at 940,638 {
   sound.playuntildone(SOUND_MENU: shadow sound.sounds_menu(field SOUND_MENU: ["part 7",":n7Gs}shamhJ6dCA:8p;"]));
 }
--- a/Stage/main.fractch
+++ b/Stage/main.fractch
@@ -9,12 +9,12 @@
 costume "backdrop1" file "assets/backdrop1.svg" center 240,180;
 sound "pop" file "assets/pop.wav" rate 48000 samples 1123;
 
-when flag at -185,346 {
-  broadcastWait "intro frame";
-  broadcastWait frame1;
-  broadcastWait frame2;
-  broadcastWait frame3;
-  broadcastWait frame4;
-  broadcastWait frame5;
-  broadcastWait frame6;
+on flag at -185,346 {
+  emit "intro frame" and wait;
+  emit frame1 and wait;
+  emit frame2 and wait;
+  emit frame3 and wait;
+  emit frame4 and wait;
+  emit frame5 and wait;
+  emit frame6 and wait;
 }
